src/main/java/com/acoustic/SpringPolandSalaryCalculator/controller/SalaryCalculatorController.java

```diff
@@ -0,0 +1,102 @@
+package com.acoustic.SpringPolandSalaryCalculator.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.acoustic.SpringPolandSalaryCalculator.calculatorservice.SalaryCalculatorService;
+import com.acoustic.SpringPolandSalaryCalculator.entity.DataSalaryCalculator;
+import com.acoustic.SpringPolandSalaryCalculator.jobcategories.JobCategories;
+import com.acoustic.SpringPolandSalaryCalculator.service.DataSalaryCalculatorRepository;
+import lombok.RequiredArgsConstructor;
+
+
+@RestController
+@RequiredArgsConstructor
+@Validated
+public class SalaryCalculatorController {
+
+    public static final int MINIMUM_GROSS = 2000;
+    private final DataSalaryCalculatorRepository dataSalaryCalculatorRepository;
+    private final List<SalaryCalculatorService> salaryCalculatorService;
+    private final JobCategories jobCategories;
+
+
+    @GetMapping("/getJobTitles/{departmentName}")
+    public String[] getJobTitles(
+            @PathVariable
+            String departmentName) {
+        return this.jobCategories.getJobDepartmentAndTitles().get(departmentName).split(",");
+    }
+
+    @GetMapping("/getJobDepartment")
+    public Set<String> getDepartmentName() {
+        return this.jobCategories.getJobDepartmentAndTitles().keySet();
+    }
+
+
+    @PostMapping("/calculator/{grossMonthlySalary}")
+    public Map<String, BigDecimal> getSalaryCalculation(
```

Comment:
name of the method could be something like calculateSalary instead of getSalaryCalculation

Reply:
done